tests/python/test_statement.py

```diff
@@ -0,0 +1,88 @@
+'''
+Tests for in-toto attestation Statement protos.
+'''
+
+import unittest
+import google.protobuf.json_format as pb_json
+
+import in_toto_attestation.v1.statement_pb2 as stpb
+import in_toto_attestation.v1.resource_descriptor_pb2 as rdpb
+
+import in_toto_attestation.v1.statement as its
+
+def create_test_statement():
+    sub = rdpb.ResourceDescriptor()
+    sub.name = 'theSub'
+    sub.digest['alg1'] = 'abc123'
+
+    st = stpb.Statement()
+    st.type = its.STATEMENT_TYPE_URI
+    st.subject.append(sub)
+    st.predicateType = 'thePredicate'
+    st.predicate.update({'keyObj': {'subKey': 'subVal'}})
+    return its.Statement(st)
+
+class TestStatement(unittest.TestCase):
+    def setUp(self):
+        self.want_st = '{"_type":"https://in-toto.io/Statement/v1","subject":[{"name":"theSub","digest":{"alg1":"abc123"}}],"predicateType":"thePredicate","predicate":{"keyObj":{"subKey":"subVal"}}}'
+
+        self.test_st = create_test_statement()
+
+    def test_create_statement(self):
+        self.test_st.validate()
+        
+    def test_json_parse_statement(self):
+        got_pb = pb_json.Parse(self.want_st, stpb.Statement())
+        got = got_pb.SerializeToString(deterministic=True)
+
+        want = self.test_st.pb.SerializeToString(deterministic=True)
+
+        self.assertEqual(got, want, 'Protos do not match')
+
+    def test_bad_statement_type(self):
+        bad_st = '{"_type":"https://in-toto.io/Statement/v0","subject":[{"name":"theSub","digest":{"alg1":"abc123"}}],"predicateType":"thePredicate","predicate":{"keyObj":{"subKey":"subVal"}}}'
```

Comment:
optional: I think this would be a bit easier to read if instead you did something like:

```
bad_st = create_test_statement()
bad_st.pb.type = 'https://in-toto.io/Statement/v0'
```

This would make it clear that the only difference from the 'passing' test and the failing one is the single change that was made.

The same approach could probably be taken for the other 'bad' tests.

WDYT?

Reply:
Yes, this makes total sense to me... not sure why I didn't think of this... thanks!

Reply:
I need to think about this more, actually. Switching to using the statement object returned by `create_test_statement()` to create bad statements would change the test setup everywhere, as we currently build the bad proto from a JSON string. And if we want to be consistent with test_resource_descriptor.py, we'd have to change the setup over there, too. Anyway, making these changes isn't hard, so I definitely don't mind doing it, if we think these changes would significantly clean up/clarify the test cases.